Library/Homebrew/cmd/list.rb

```diff
@@ -116,8 +116,14 @@ def list
       ls_args << "-r" if args.r?
       ls_args << "-t" if args.t?
 
-      safe_system "ls", *ls_args, HOMEBREW_CELLAR unless args.cask?
-      list_casks(args: args) unless args.formula?
+      unless args.cask?
```

Comment:
It turns out this check is redundant, because of:

https://github.com/Homebrew/brew/blob/a045cd7a614246a2debc537e9cd406ab55316743/Library/Homebrew/cmd/list.rb#L106-L107